File: app/src/main/java/com/imangazalievm/bubbble/domain/interactors/UserFollowersInteractor.java
package com.imangazalievm.bubbble.domain.interactors;


import com.imangazalievm.bubbble.domain.models.Follow;
import com.imangazalievm.bubbble.domain.models.User;
import com.imangazalievm.bubbble.domain.models.UserFollowersRequestParams;
import com.imangazalievm.bubbble.domain.repository.FollowersRepository;
import com.imangazalievm.bubbble.domain.repository.UsersRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Single;

public class UserFollowersInteractor {

    private UsersRepository usersRepository;
    private FollowersRepository followersRepository;

    @Inject
    public UserFollowersInteractor(UsersRepository usersRepository, FollowersRepository followersRepository) {
        this.usersRepository = usersRepository;
        this.followersRepository = followersRepository;
    }

    public Single<User> getUser(long userId) {
        return usersRepository.getUser(userId);
    }

    public Single<List<Follow>> getUserFollowers(UserFollowersRequestParams requestParams) {
        return followersRepository.getUserFollowers(requestParams);
    }

}

File: app/src/main/java/com/bubbble/domain/userprofile/UserFollowersInteractor.kt
package com.bubbble.domain.userprofile

import com.bubbble.core.models.user.Follow
import com.bubbble.data.users.FollowersRepository
import com.bubbble.core.models.user.UserFollowersParams
import javax.inject.Inject

class UserFollowersInteractor @Inject constructor(
    private val followersRepository: FollowersRepository
) {

    suspend fun getUserFollowers(params: UserFollowersParams): List<Follow> {
        return followersRepository.getUserFollowers(params)
    }

}